component/src/main/java/io/siddhi/extension/io/file/FileHandler.java

```diff
@@ -0,0 +1,409 @@
+                                "which has been  modified or created or deleted in the execution time " +
+                                "will be identified and displayed. \n After monitoring has been started," +
+                                " If any files in the specific uri is newly created, modified or deleted.\n " +
+                                "an event is created with the filepath, length, last modified date " +
+                                "and status of the file. Then that will be received by the FooStream. If there " +
+                                "are any changes a new event will be generated in every 200 milliseconds.\n"
+                ),
+
+        }
+)
+
+public class FileHandler extends Source<FileHandler.FileHandlerState> {
+    private static final Logger log = Logger.getLogger(FileHandler.class);
+    private SourceEventListener sourceEventListener;
+    private long monitoringInterval = 500;
+    private String uri;
+    private String listeningDirUri;
+    private String listeningFileUri;
+    private FileAlterationMonitor monitor;
+    private Map<String, Long> initialMap = new ConcurrentHashMap<>();
+    private SiddhiAppContext siddhiAppContext;
+    private List<SubThread> threadList = new ArrayList<>();
+    private static final String CURRENT_MAP_KEY = "current.map.key";
+
+    @Override
+    protected ServiceDeploymentInfo exposeServiceDeploymentInfo() {
+        return null;
+    }
+
+    @Override
+    public StateFactory<FileHandlerState> init(SourceEventListener sourceEventListener,
+                                               OptionHolder optionHolder,
+                                               String[] requiredProperties,
+                                               ConfigReader configReader,
+                                               SiddhiAppContext siddhiAppContext) throws SiddhiAppValidationException {
+        this.siddhiAppContext = siddhiAppContext;
+        this.sourceEventListener = sourceEventListener;
+        if (optionHolder.isOptionExists(Constants.URI)) {
+            uri = optionHolder.validateAndGetStaticValue(Constants.URI);
+            FileObject file = Utils.getFileObject(uri);
+            try {
+                if (!file.exists()) {
+                    throw new SiddhiAppCreationException(" Directory/File " + file.getPublicURIString()
+                            + " is not found ");
+                }
+                if (file.isFile()) {
+                    listeningFileUri = file.getName().getPath();
+                    file = file.getParent();
+                }
+                listeningDirUri = file.getName().getPath();
+            } catch (FileSystemException e) {
+                throw new SiddhiAppValidationException("Directory/File " + file.getPublicURIString()
+                        + " is not found.", e);
+            }
+        }
+        if (uri == null) {
```

Comment:
This has to be checked in the beginning before 'FileObject file = Utils.getFileObject(uri);"

Reply:
fixed in 3f9a2a0